scripts/ci/build.sh

```diff
@@ -2,17 +2,28 @@
 set -ex
 
 ROOTDIR="$(dirname "$0")/../.."
+# shellcheck disable=SC1091
+source "${ROOTDIR}/scripts/common.sh"
+
+prerelease_source="$1"
+
 cd "${ROOTDIR}"
 
 # shellcheck disable=SC2166
 if [ "$CIRCLE_BRANCH" = release -o -n "$CIRCLE_TAG" -o -n "$FORCE_RELEASE" ]
 then
-    echo -n "" >prerelease.txt
+    if [[ -n "$prerelease_source" ]]; then
+        fail "prerelease_source must not be set when building a release"
+    fi
```

Comment:
I'd generally make these checks a little less verbose, at least in cases where it's just a check and a fail with a message:
```suggestion
    [[ -n "$prerelease_source" ]] || fail "prerelease_source must not be set when building a release"
```